pkg/cdi/cdi.go

```diff
@@ -80,12 +84,26 @@ func (c *impl) CreateCDISpecForPool(resourcePrefix string, rPool types.ResourceP
 		cdiSpec.Devices = append(cdiSpec.Devices, device)
 	}
 
-	err = cdi.GetRegistry().SpecDB().WriteSpec(&cdiSpec, cdiSpecPrefix+resourcePrefix)
+	// calculate hash of the cdiSpec
+	digest, err := extractEncodedFromDigest(Digest(cdiSpec).String())
+	if err != nil {
+		glog.Errorf("CreateCDISpecForPool(): can not calculate hash of the cdiSpec: %v", err)
+		return err
+	}
+	name, err := cdi.GenerateNameForTransientSpec(&cdiSpec, digest)
+	if err != nil {
+		glog.Errorf("CreateCDISpecForPool(): can not generate transient name: %v", err)
+		return err
+	}
+
+	err = cdi.GetRegistry().SpecDB().WriteSpec(&cdiSpec, cdiSpecPrefix+name)
```

Comment:
can we use  cdiSpecPrefix+ the pool name here ?

e.g :  `"sriov-dp-mellanox.com-rdma_resource_1"` 

that way we write the same file  then, no need to worry about c.lastGenratedName

Reply:
I went with this to assure a unique name. There is no guarantee for the pool name to be unique right?

Reply:
we can assume that same as we assume no one else will create device plugin (unix) socket per resoruce. under kubelet dir (/var/lib/kubelet/plugins_registry/).

moreover for debuggability it would be better to understand which file belongs to which resource.